addons/mail/static/src/core/web/thread_service_patch.js

```diff
@@ -227,6 +227,25 @@ patch(ThreadService.prototype, "mail/core/web", {
         }
         this._super(...arguments);
     },
+    getRecentThreads(allowedTypes = ["channel", "chat", "group"]) {
+        return Object.values(this.store.threads)
+            .filter((thread) => allowedTypes.includes(thread.type))
+            .sort((a, b) => {
+                if (!a.lastInterestDateTime && !b.lastInterestDateTime) {
+                    return 0;
+                }
+                if (a.lastInterestDateTime && !b.lastInterestDateTime) {
+                    return -1;
+                }
+                if (!a.lastInterestDateTime && b.lastInterestDateTime) {
+                    return 1;
+                }
+                return b.lastInterestDateTime.ts - a.lastInterestDateTime.ts;
+            });
+    },
+    getMentionedThreads() {
+        return this.getRecentThreads().filter((thread) => thread.message_needaction_counter > 0);
+    },
```

Comment:
I am not sure if this is what we want because, in the spec, only mentions are counted. But indeed, we use `message_needaction_counter` so I will just rename the method into this.

Reply:
Specs state mentions because that's the most frequent when it comes from chatter or channels. But in practice all unread messages in chats are equivalent to needactions. The counter in sidebar and messaging menu reflects that, and we expect the exact same showing of counter in the command palette.
Therefore the counter you need is `getCounter()` in thread service. `message_needaction_counter` will only take chatter / channels into account but not chats, and chats should definitely be taken into account